Allow TodoistAPI date_string in service call

bjw-s opened the pull request:
## Description:
In the original implementation of the Todoist calendar component it was not possible to allow Todoist natural language for due dates. This PR adds this capability.

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#4923

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

bjw-s committed:
Support date_string in service call

andreipop2005 committed:
Edimax component reports wrong power values (#13011)

* Fixed Edimax switch authentication error for newer firmware.

* pyedimax moved to pypi

* Added pyedimax to gen_requirements_all.py

* Cleanup

* Fixed https://github.com/home-assistant/home-assistant/issues/13008

* Only ValueError now

* Trivial error.

the-glu committed:
myStrom: Add RGB support to Wifi bulbs (#13194)

balloob committed:
Fix logbook JSON serialize issue (#13229)

* Fix logbook JSON serialize issue

* Address flakiness

* Lint

* deflake ?

* Deflake 2

maxlaverse committed:
Try to fix caldav (#13236)

* Fix device attribute type for event end

* Fix is_over and add tests

bjw-s committed:
Add extra parameters to service schema

Juggels commented:
I wasn't sure how to make sure that the change from home-assistant/home-assistant#13215 don't get lost. To be safe, I have **not** included that change in this PR.

amelchio committed:
Cleanup Sonos platform setup (#13225)

* Cleanup Sonos platform setup

* Remove unneeded lists

bjw-s committed:
Fix 'dict' object has no attribute 'strftime' (#13215)

* Fix 'dict' object has no attribute 'strftime'

* Clear existing list instead of new object

balloob committed:
Fix aiohttp deprecation warnings (#13240)

* Fix aiohttp deprecation warnings

* Fix Ring deprecation warning

* Lint

balloob committed:
Tado: don't reference unset hass var (#13237)

Tado: don't reference unset hass var

gerard33 committed:
Add extra sensors for BMW ConnectedDrive (#12591)

* Added extra sensors for BMW ConnectedDrive

* Updates based on review of @MartinHjelmare

* Updates based on 2nd review of @MartinHjelmare

* Changed control flow for updates to support updates triggered by remote services.

* updated library version number

* Changed order of commands so that the UI looks consistent.

State of lock is now set optimisitcally before getting proper update
from the server. So that the state does not toggle in the UI.

* Added comment on optimistic state

* Updated requirements_all.txt

* Revert access permission changes

* Fix for Travis

* Changes based on review by @MartinHjelmare

amelchio committed:
Fix Sonos join/unjoin in scripts (#13248)

cdce8p committed:
Homekit: New supported devices (#13244)

* Fixed log message
* Added support for scripts
* Added support for lights
* Small refactoring
* Added support for humidity sensor
* Added tests

reefab committed:
Adding Foobot device sensor (#12417)

* Added Foobot device sensor

* Added error handling tests

* Corrections after PR review.

* Migrated to async/await syntax

 * lint fixes
 * stop raising HomeAssistantError
 * debug log for number of requests

* Moved shared data between sensors from a class attribute to a separate class

* Made test more async-aware

disabled setup error test for now as it's not working

* Working failure scenarios tests

karlkar committed:
Fix for not setting up the camera if it is offline during setup phase (#13082)

* Fix for not setting up the camera if it is offline during setup phase

* async/await and modified service creation

* Properly handle not supported PTZ

* setup platform made synchronous as ONVIFService constructors do I/O

* Fix intendation issue

BioSehnsucht committed:
Added Stride notification component (#13221)

* Added Stride notification component

* Fix trailing whitespace in Stride notify

* More whitespace fixes and rogue comment for Stride notify

* More whitespace fixing for Stride notify

* Correcting hanging indents for Stride notify

Juggels commented:
Messed this one up by updating my source branch, will open a new one
